alerting/src/main/kotlin/org/opensearch/alerting/DocumentLevelMonitorRunner.kt

```diff
@@ -134,73 +138,84 @@ object DocumentLevelMonitorRunner : MonitorRunner() {
 
             // cleanup old indices that are not monitored anymore from the same monitor
             for (ind in updatedLastRunContext.keys) {
-                if (!indices.contains(ind)) {
+                if (!concreteIndices.contains(ind)) {
                     updatedLastRunContext.remove(ind)
                 }
             }
 
             // Map of document ids per index when monitor is workflow delegate and has chained findings
             val matchingDocIdsPerIndex = workflowRunContext?.matchingDocIdsPerIndex
 
-            indices.forEach { indexName ->
-                // Prepare lastRunContext for each index
-                val indexLastRunContext = lastRunContext.getOrPut(indexName) {
-                    val isIndexCreatedRecently = createdRecently(
-                        monitor,
-                        periodStart,
-                        periodEnd,
-                        monitorCtx.clusterService!!.state().metadata.index(indexName)
-                    )
-                    MonitorMetadataService.createRunContextForIndex(indexName, isIndexCreatedRecently)
-                }
-
-                // Prepare updatedLastRunContext for each index
-                val indexUpdatedRunContext = updateLastRunContext(
-                    indexLastRunContext.toMutableMap(),
-                    monitorCtx,
-                    indexName
-                ) as MutableMap<String, Any>
-                updatedLastRunContext[indexName] = indexUpdatedRunContext
-
-                val count: Int = indexLastRunContext["shards_count"] as Int
-                for (i: Int in 0 until count) {
-                    val shard = i.toString()
-
-                    // update lastRunContext if its a temp monitor as we only want to view the last bit of data then
-                    // TODO: If dryrun, we should make it so we limit the search as this could still potentially give us lots of data
-                    if (isTempMonitor) {
-                        indexLastRunContext[shard] = max(-1, (indexUpdatedRunContext[shard] as String).toInt() - 10)
+            docLevelMonitorInput.indices.forEach { indexName ->
```

Comment:
why did we change this list being looped from computed concrete indices list to the monitor indices list?

Reply:
this is because we want to expand `1 index pattern` & process all its `concrete indices` separately. Previous logic was we expand `all index patterns` & process all the `concrete indices` one by one.